param_passing/proposal.md

```diff
@@ -0,0 +1,388 @@
+approach was chosen so that downstream users can choose to ignore the namespaced
+custom element and its contents. The goal is to implement passing additional
+arguments in `//include` through an experimental custom element in
+SDFormat 1.7 / `libsdformat` 10 then after being vetted it will be made official
+in a future SDFormat / `libsdformat` release where the custom element will be
+changed to `//include/params`.The included model (not the constructed model in
+the model/world file) will be referred to as the original model/file.
+In the model/world file and under `//include`, the elements listed under
+`//include/experimental:params` will indicate elements from the original model
+and will specify new values and/or elements to be updated, added, and/or
+removed.
+
+**Details**
+
+To specify an element from the original file, the tag (e.g., `//link`,
+`//sensor`, `//visual`) needs to be provided as well as the `element_id` attribute
+where the `element_id` is the name of all the parent elements leading to the
+specified element separated by double colons (`::`). For example, given:
+
+```xml
+<link name="chassis">
+  <sensor name="camera">
+    <camera name="cam">
+      ...
+    </camera>
+  </sensor>
+</link>
+```
+
+If the user wants to specify the `//camera` element, then
+`element_id="chassis::camera::cam"` where "chassis" is the name of the `//link`,
+"camera" is the name of the `//link/sensor`, and "cam" is the name of the
+`//link/sensor/camera`. This is how the correct element will be identified and
+will be called the element identifier. Although it is possible to identify the
+element without requiring the user to specify the tag, enforcing the user to
+provide the tag allows less ambiguity to the user as well as provides a means of
+error checking. Then the user will need to specify a corresponding action with
+that element to dictate the alteration they would like to make using a new
+attribute `action`. The attribute takes in a string and the available actions
+are:
+
+* `modify`: indicates that the values in the original model are to be modified
+to the new listed values
+* `replace`: replaces the elements and/or values from the original model to the
+new provided elements and/or values
+* `add`: adds new elements to the original model
+* `remove`: removes (or disables) the elements from the original model
+
+The `action` attribute must be provided either in the element identifier or in
+the direct children of the identifier.
+
+##### Alternatives considered
+
+Alternatively, instead of using the proposed element identifier, the user could
+use the attribute `id` for each customizable element in the original file but
+the intent of using the named path (i.e., the name of the parent elements and
+itself separated by double colons) was to provide a way for the user to modify
+the original model without requiring large amounts of updates to the original
+file.
+
+## Examples
```

Comment:
Is it possible to add an example for modifying the inertia and CoM of a link?

Reply:
7c8d6438fab8772a89119f927cdc145cf2948772